ttnn/cpp/ttnn/operations/moreh/moreh_linear/moreh_linear.hpp

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/cpp/ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+
+namespace ttnn::operations::moreh::moreh_linear {
+struct MorehLinear {
+    static std::optional<Tensor> invoke(
```

Comment:
Why is return optional here?

Reply:
When optional preallocated output is provided, the returned tensor should be that preallocated tensor.

Reply:
resolved in [d58bf0a](https://github.com/tenstorrent/tt-metal/pull/12812/commits/d58bf0af9e7b737fe195b277b10a0d4da4bd49c1)